pkg/kn/commands/service/create.go

```diff
@@ -303,12 +303,11 @@ func constructServiceFromFile(cmd *cobra.Command, editFlags ConfigurationEditFla
 	// Set namespace in case it's specified as --namespace
 	service.ObjectMeta.Namespace = namespace
 
-	// We need to generate revision to have --force replace working
-	revName, err := servinglib.GenerateRevisionName(editFlags.RevisionName, &service)
+	// Apply options provided from cmdline
+	err = editFlags.Apply(&service, nil, cmd)
 	if err != nil {
 		return nil, err
 	}
```

Comment:
I could comment these lines and all tests are still passing. So this tells me no test for errors when `editFlags.Apply(...)`

Reply:
But isn't `editFlags.Apply()` already covered ? Ideally, there is a dedicated unit test only for this method, and should not be tested indirectly within tests that test that intend to test other stuff.

Reply:
If I look at the overall test suite I see, that `editFlags.Apply()` is already tested to a high degree. Looking at that method though it looks like that it deserves some refactoring, it is far too large.